### ISSUE TITLE
Lost the formatting method call in checked field getter

### ISSUE THREAD

**Issue by bin1357** [final-form/react-final-form#858]
### Are you submitting a **bug report** or a **feature request**?
**bug report**
Lost the formatting method call in getting the checked field value
https://github.com/final-form/react-final-form/blob/master/src/useField.js#L221
it is impossible to make a checkbox that would switch between two true values. For example 'on' and 'off'

### What is the current behavior?
it is impossible

### What is the expected behavior?
it is possible

### Sandbox Link
https://codesandbox.io/s/keen-oskar-l1vku

**Comment by bin1357** [comment]
https://codesandbox.io/s/final-formreact-final-form-simple-forked-f4j86 fixed

**Comment by bin1357** [comment]
```
<Field
    name="employed"
    parse={(t) => (t ? "on" : "off")}
    format={(f) => (f === "on" ? true : false)}
    component="input"
    type="checkbox"
/>
```

**Comment by ArturKp** [comment]
I'm not familiar with the codebase, so can't say anything about the code changes, but this seems to also fix my odd usecase, which I was expecting to work.

```
parse={t => t ? null : 1}
format={f => f === null}
```

Checked: null
Unchecked: 1

**Comment by erikras** [comment]
Published fix in [`v6.5.2`](https://github.com/final-form/react-final-form/releases/tag/v6.5.2).
